tests/cra.test.js

```diff
@@ -31,86 +32,83 @@ test.after(async (t) => {
+  const { body, statusCode, headers } = await got(`${url}/test.html`)
 
-  t.is(response.status, 200)
-  t.truthy(response.headers.get('content-type').startsWith('text/html'))
+  t.is(statusCode, 200)
+  t.true(headers['content-type'].startsWith('text/html'))
   t.true(body.includes('<h1>Test content</h1>'))
 })
 
 test('redirect test', async (t) => {
   const { url } = t.context.server
-  const response = await fetch(`${url}/something`, { redirect: 'manual' })
+  const { body, statusCode, headers } = await got(`${url}/something`, { followRedirect: false })
 
-  t.is(response.status, 301)
-  t.is(response.headers.get('location'), `${url}/otherthing.html`)
-  t.is(await response.text(), 'Redirecting to /otherthing.html')
+  t.is(statusCode, 301)
+  t.is(headers.location, `/otherthing.html`)
+  t.is(body, 'Redirecting to /otherthing.html')
 })
 
 test('normal rewrite', async (t) => {
   const { url } = t.context.server
+  // TODO: replace with got like the rest of the tests
+  // This test passes with fetch and curl, but not with got (returns 404 with got)
   const response = await fetch(`${url}/doesnt-exist`)
   const body = await response.text()
 
   t.is(response.status, 200)
-  t.truthy(response.headers.get('content-type').startsWith('text/html'))
-  t.regex(body, /Web site created using create-react-app/)
+  t.true(response.headers.get('content-type').startsWith('text/html'))
+  t.true(body.includes('Web site created using create-react-app'))
 })
 
 test('force rewrite', async (t) => {
   const { url } = t.context.server
-  const response = await fetch(`${url}/force.html`)
-  const body = await response.text()
+  const { body, statusCode, headers } = await got(`${url}/force.html`)
 
-  t.is(response.status, 200)
-  t.truthy(response.headers.get('content-type').startsWith('text/html'))
+  t.is(statusCode, 200)
+  t.true(headers['content-type'].startsWith('text/html'))
   t.true(body.includes('<h1>Test content</h1>'))
 })
 
 test('robots.txt', async (t) => {
   const { url } = t.context.server
-  const response = await fetch(`${url}/robots.txt`)
-  const body = await response.text()
+  const { body, statusCode, headers } = await got(`${url}/robots.txt`)
 
-  t.is(response.status, 200)
-  t.truthy(response.headers.get('content-type').startsWith('text/plain'))
-  // First line of the file
-  t.regex(body, /# https:\/\/www.robotstxt.org\/robotstxt.html/)
+  t.is(statusCode, 200)
+  t.true(headers['content-type'].startsWith('text/plain'))
+  t.true(body.startsWith('# https://www.robotstxt.org/robotstxt.html'))
 })
 
 test('functions rewrite echo without body', async (t) => {
   const { url, host, port } = t.context.server
-  const response = await fetch(`${url}/api/echo?ding=dong`).then((res) => res.json())
+  const response = await got(`${url}/api/echo?ding=dong`).json()
 
   t.is(response.body, undefined)
   t.deepEqual(response.headers, {
-    accept: '*/*',
-    'accept-encoding': 'gzip,deflate',
+    accept: 'application/json',
```

Comment:
Some of these values can be take from the `got` request object.
I find it easier to have the expected values in plain strings.